File: bindings/java-ffm/src/main/java/org/maplibre/nativeffi/StyleImageOptions.java
package org.maplibre.nativeffi;

/** Mutable descriptor for runtime style image options. */
public final class StyleImageOptions {
  private Float pixelRatio;
  private Boolean sdf;

  public boolean hasPixelRatio() {
    return pixelRatio != null;
  }

  public Float pixelRatio() {
    return pixelRatio;
  }

  public StyleImageOptions setPixelRatio(float pixelRatio) {
    if (!Float.isFinite(pixelRatio) || pixelRatio <= 0.0f) {
      throw new IllegalArgumentException("pixelRatio must be finite and positive");
    }
    this.pixelRatio = pixelRatio;
    return this;
  }

  public StyleImageOptions clearPixelRatio() {
    pixelRatio = null;
    return this;
  }

  public boolean hasSdf() {
    return sdf != null;
  }

  public Boolean sdf() {
    return sdf;
  }

  public StyleImageOptions setSdf(boolean sdf) {
    this.sdf = sdf;
    return this;
  }

  public StyleImageOptions clearSdf() {
    sdf = null;
    return this;
  }
}

File: bindings/kotlin-native/src/nativeMain/kotlin/org/maplibre/nativeffi/style/StyleImageOptions.kt
package org.maplibre.nativeffi.style

/** Mutable descriptor for runtime style image options. */
public class StyleImageOptions {
  public var pixelRatio: Float? = null
    private set

  public var sdf: Boolean? = null
    private set

  public fun hasPixelRatio(): Boolean = pixelRatio != null

  public fun pixelRatio(pixelRatio: Float): StyleImageOptions = apply {
    this.pixelRatio = pixelRatio
  }

  public fun clearPixelRatio(): StyleImageOptions = apply { pixelRatio = null }

  public fun hasSdf(): Boolean = sdf != null

  public fun sdf(sdf: Boolean): StyleImageOptions = apply { this.sdf = sdf }

  public fun clearSdf(): StyleImageOptions = apply { sdf = null }
}
